webhook.go

```diff
@@ -0,0 +1,210 @@
+		if errResp != nil {
+			log.Errorf("try %d, return health check of the rest service for error  %v", i, errResp)
+			time.Sleep(healthCheckPause)
+			err = errResp
+			continue
+		}
+		if response != nil && response.StatusCode == http.StatusOK {
+			response.Body.Close()
+			log.Infof("rest service returned healthy status")
+			time.Sleep(healthCheckPause)
+			err = nil
+			return
+		}
+		response.Body.Close()
+	}
+	if err != nil {
+		err = fmt.Errorf("error connecting to rest api %s", err.Error())
+	}
+	return
+}
+
+// Port port id
+func (s *Webhook) Port() int {
+	return s.port
+}
+
+//Ready gives the status of the server
+func (s *Webhook) Ready() bool {
+	return s.status == started
+}
+
+// GetHostPath  returns hostpath
+func (s *Webhook) GetHostPath() *types.URI {
+	return types.ParseURI(fmt.Sprintf("http://localhost:%d%s", s.port, s.apiPath))
+}
+
+// Start will start res routes service
+func (s *Webhook) Start() {
+	if s.status == started || s.status == starting {
+		log.Infof("Server is already running at port %d", s.port)
+		return
+	}
+	s.status = starting
+	r := mux.NewRouter()
+	api := r.PathPrefix(s.apiPath).Subrouter()
+
+	api.HandleFunc("/webhook", s.publishHwEvent).Methods(http.MethodPost)
+
+	err := r.Walk(func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error {
+		pathTemplate, err := route.GetPathTemplate()
+		if err == nil {
+			log.Println("ROUTE:", pathTemplate)
+		}
+		pathRegexp, err := route.GetPathRegexp()
+		if err == nil {
+			log.Println("Path regexp:", pathRegexp)
+		}
+		queriesTemplates, err := route.GetQueriesTemplates()
+		if err == nil {
+			log.Println("Queries templates:", strings.Join(queriesTemplates, ","))
+		}
+		queriesRegexps, err := route.GetQueriesRegexp()
+		if err == nil {
+			log.Println("Queries regexps:", strings.Join(queriesRegexps, ","))
+		}
+		methods, err := route.GetMethods()
+		if err == nil {
+			log.Println("Methods:", strings.Join(methods, ","))
+		}
+		log.Println()
+		return nil
+	})
+
+	if err != nil {
+		log.Println(err)
+	}
+	api.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, r)
+	})
+
+	log.Info("starting rest api server")
+	log.Infof("endpoint %s", s.apiPath)
+	go wait.Until(func() {
+		s.status = started
+		s.httpServer = &http.Server{
+			Addr:    fmt.Sprintf(":%d", s.port),
+			Handler: api,
+		}
+		err := s.httpServer.ListenAndServe()
+		if err != nil {
+			log.Errorf("restarting due to error with api server %s\n", err.Error())
+			s.status = failed
+		}
+	}, 1*time.Second, s.closeCh)
+}
+
+// Shutdown ... shutdown rest service api, but it will not close until close chan is called
+func (s *Webhook) Shutdown() {
+	log.Warnf("trying to shutdown rest api sever, please use close channel to shutdown ")
+	s.httpServer.Close()
+}
+
+// publishHwEvent gets redfish HW events and converts it to cloud native event and publishes to the hw publisher
+func (s *Webhook) publishHwEvent(w http.ResponseWriter, r *http.Request) {
```

Comment:
HW webhook is designed to receive the redfish event and then publish it to the transport bus. So in this case, call default pubsub API which is part of the sidecar. We need to make this take both Hw event data and cloud events data schema.
url := &types.URI{URL: url.URL{Scheme: "http",
		Host: apiAddr,
		Path: fmt.Sprintf("%s%s", apiPath, "create/event")}}